src/main/java/com/sap/sgs/phosphor/fosstars/data/github/FirstCommit.java

```diff
@@ -55,13 +52,13 @@ public FirstCommit update(ValueSet values) throws IOException {
    * @return An instance of {@link Value} which contains a date of the first commit.
    * @throws IOException If something went wrong.
    */
-  Value<Date> firstCommitDate() throws IOException {
-    Optional<Value> something = cache().get(url, FIRST_COMMIT_DATE);
+  Value<Date> firstCommitDate(GitHubProject project) throws IOException {
+    Optional<Value> something = cache().get(project, FIRST_COMMIT_DATE);
     if (something.isPresent()) {
       return something.get();
```

Comment:
this cached return value will be stored in cache again.
- What if there is a request for the same project every day and it always finds something in cache. Then the first commitDate would never be updated.

Reply:
Good catch! The cache should be updated in the end of `firstCommitDate()` method if a new value was calculated. I'll update it, thanks!